File: Modules/HumanResource/AnnualLeavePolicy/AnnualLeavePolicy/AnnualLeavePolicyForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.AnnualLeavePolicy.Forms;

[FormScript("AnnualLeavePolicy.AnnualLeavePolicy")]
[BasedOnRow(typeof(AnnualLeavePolicyRow), CheckNames = true)]
public class AnnualLeavePolicyForm
{
    //public int Year { get; set; }
    public int ServiceFromYear { get; set; }
    public int ServiceUntilYear { get; set; }
    public int EligibleDays { get; set; }
    public int MaximumAccumulated { get; set; }
}

File: Modules/ServerTypes/AnnualLeavePolicy/AnnualLeavePolicyForm.ts
import { IntegerEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface AnnualLeavePolicyForm {
    ServiceFromYear: IntegerEditor;
    ServiceUntilYear: IntegerEditor;
    EligibleDays: IntegerEditor;
    MaximumAccumulated: IntegerEditor;
}

export class AnnualLeavePolicyForm extends PrefixedContext {
    static formKey = 'AnnualLeavePolicy.AnnualLeavePolicy';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!AnnualLeavePolicyForm.init)  {
            AnnualLeavePolicyForm.init = true;

            var w0 = IntegerEditor;

            initFormType(AnnualLeavePolicyForm, [
                'ServiceFromYear', w0,
                'ServiceUntilYear', w0,
                'EligibleDays', w0,
                'MaximumAccumulated', w0
            ]);
        }
    }
}